libbeat/autodiscover/providers/providers.go

```diff
@@ -0,0 +1,38 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package providers
+
+import (
+	"github.com/elastic/beats/v7/libbeat/autodiscover"
+	"github.com/elastic/beats/v7/libbeat/autodiscover/providers/docker"
+	"github.com/elastic/beats/v7/libbeat/autodiscover/providers/jolokia"
+	"github.com/elastic/beats/v7/libbeat/autodiscover/providers/kubernetes"
+)
+
+var KnownProviders = map[string]autodiscover.ProviderBuilder{
+	docker.ProviderName:     docker.AutodiscoverBuilder,
+	jolokia.ProviderName:    jolokia.AutodiscoverBuilder,
+	kubernetes.ProviderName: kubernetes.AutodiscoverBuilder,
+}
+
+func AddKnownProviders(r *autodiscover.Registry, providers map[string]autodiscover.ProviderBuilder) error {
+	for name, builder := range providers {
+		_ = r.AddProvider(name, builder)
+	}
+	return nil
+}
```

Comment:
I'm confused here... Everything can return an error, but they're always ignored... So either `AddKnownProviders` should not return an error, or the errors should be joined with `errors.Join` and returned.

Reply:
thanks fixed.